src/pages/index.js

```diff
@@ -0,0 +1,312 @@
+const deleteFormCancelBtn = deleteModal.querySelector(".modal__cancel-btn");
+
+// Load Images
+document.getElementById("pencil").src = pencilSrc;
+document.getElementById("logo").src = logoSrc;
+document.getElementById("plus").src = plusSrc;
+document.getElementById("bessie-coleman").src = colemanSrc;
+
+// Utility Function
+function disableButton(buttonElement, settings) {
+  buttonElement.disabled = true;
+  buttonElement.classList.add(settings.inactiveButtonClass);
+}
+
+// Handle Add Card
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const inputValues = {
+    name: cardNameInput.value,
+    link: cardLinkInput.value,
+  };
+
+  setButtonText(evt.submitter, "Saving...");
+
+  api
+    .addCard(inputValues)
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardList.prepend(cardElement);
+      closeModal(cardModal);
+      evt.target.reset();
+      disableButton(cardSubmitButton, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(evt.submitter, false, "Saving...", "Save");
+    });
+}
+
+// Handle Like Card
+function handleLike(evt, id) {
+  const isLiked = evt.target.classList.contains("card__like-btn_liked");
+  evt.target.classList.toggle("card__like-btn_liked");
+  api
+    .changeLikeStatus(id, isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__like-btn_active");
+    })
+    .catch(console.error);
+}
+
+// Handle Delete Card
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(evt.submitter, true, "Delete", "Deleting..."); // Set loading state
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      selectedCard = null;
+      selectedCardId = null;
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => setButtonText(evt.submitter, false, "Delete")); // Reset button text
+}
+
+// Handle Edit Profile
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  setButtonText(evt.submitter, true, "Save", "Saving..."); // Set loading state
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => setButtonText(evt.submitter, false, "Save")); // Reset button text
+}
+
+// Handle Avatar Update
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(evt.submitter, "Saving...");
+
+  api
+    .editAvatarInfo(avatarLinkInput.value)
+    .then((data) => {
+      console.log(data);
+      if (data.avatar) {
+        document.querySelector(".profile__avatar").src = data.avatar;
```

Comment:
`document.querySelector(".profile__avatar")` 

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine

Reply:
My remark hasn't been corrected https://github.com/BillyChilly808/se_project_spots/pull/4/files#r1969249356